### PR TITLE
[4.0] Respect stage ordering

### PR DESCRIPTION
**Pull request by bembelimen**
Pull Request for Issue #21537 .

### Summary of Changes
Ordering of stages are respected now

### Testing Instructions
Reorder stages and check in articles filter (backend) or transition edit view if the ordering of the stages are correct.

### CODE REVIEW

**Comment by ghost** [comment]
changed Title as its about 4.0.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21705">issues.joomla.org/tracker/joomla-cms/21705</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :red_circle: unsuccessfully on 503b7fac08d345721fd3c2a648a1f951a769f265<br /><br />Looks like ordering of Transitions are shown, not of Stages. This works with and -out Pull Request.
#### Ordering shows Transitions
screen shot 2018-08-19 at 07 43 31
#### Ordering Transitions
screen shot 2018-08-19 at 07 43 53
#### Ordering Stages
screen shot 2018-08-19 at 07 44 08
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21705">issues.joomla.org/tracker/joomla-cms/21705</a>.</sub>

File: libraries/src/Form/Field/WorkflowstageField.php [hunk 1]
```diff
@@ -98,7 +98,8 @@ protected function getGroups()
 				->from($db->quoteName('#__workflows', 'w'))
 				->where($db->quoteName('ws.workflow_id') . ' = ' . $db->quoteName('w.id'))
 				->where($db->quoteName('w.extension') . ' = ' . $db->quote($this->extension))
-				->order($db->quoteName('w.ordering'));
+				->order($db->quoteName('w.ordering'))
+				->order($db->quoteName('ws.ordering'));
```
Comment (wilsonge) [inline libraries/src/Form/Field/WorkflowstageField.php:107]: I don't think you need order here twice :) (I get this isn't a duplicate but I don't think you can order on both)

Comment (bembelimen) [reply libraries/src/Form/Field/WorkflowstageField.php:107]: The first ordering is for the workflow itself, so the groups are ordered correctly and the second ordering for the stages, so the values are ordered?

Or I didn't get your point :)

**Comment by bembelimen** [comment]
@franz-wohlkoenig you're looking on the wrong page:

> Reorder stages and check in articles filter (backend) or transition edit view if the ordering of the stages are correct.

**Comment by ghost** [comment]
I have tested this item :red_circle: unsuccessfully on df1f3aad1ae0a88fe74d3b740c77ab562cf2455d

#### Reordered Stages:
screen shot 2018-08-24 at 07 05 15
#### Transistion Edit View: Successfully
screen shot 2018-08-24 at 07 04 57
#### Articles Filter: Unsuccessfully
screen shot 2018-08-24 at 07 04 15


<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21705">issues.joomla.org/tracker/joomla-cms/21705</a>.</sub>

**Comment by bembelimen** [comment]
@franz-wohlkoenig your last image are transitions, not stages

**Comment by ghost** [comment]
> and check in articles filter (backend)

i thought you mean the Dropdownfield. What to look for, the "Select Condition" in "Table Options" or "Select State"?

**Comment by bembelimen** [comment]
grafik

grafik

**Comment by bembelimen** [comment]
Does it work for you now @franz-wohlkoenig ?

**Comment by ghost** [comment]
- Status in Article-Edit (your second Screenshot) works.
- Table-Options in Article-View show:

screen shot 2018-09-05 at 08 46 09

Please Note "- Select **State** -" which is different to your first Screenshot too.
### System information
- Nightly Build 4.0.0-alpha5-dev
- Multilingual Sample Data (French, German DE, Persian)
- Template: Cassiopeia
- macOS Sierra, 10.13.6
- Firefox 61 (64-bit)
#### CloudAccess.net
- PHP 7.1.19
- MySQLi 5.7.21-cll-lve

**Comment by baernm** [comment]
I have tested this item :white_check_mark: successfully on 5061bd906173433a1fadf3705944296c4e847b54<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21705">issues.joomla.org/tracker/joomla-cms/21705</a>.</sub>

**Comment by bembelimen** [comment]
@franz-wohlkoenig I just have more workflow and stages in my system :) (you need at least 2 stages to sort them)

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**